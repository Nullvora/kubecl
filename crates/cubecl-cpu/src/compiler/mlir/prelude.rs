#![allow(unused)]

pub(super) use super::item::TransformItem;
pub(super) use super::types::TransformType;

pub(super) use melior::Context;
