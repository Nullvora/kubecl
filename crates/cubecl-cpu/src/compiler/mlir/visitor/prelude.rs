pub use super::Visitor;
use tracel_llvm::melior::{Context, ir::Type};

pub trait IntoType {
    fn to_type<'a>(self, context: &'a Context) -> Type<'a>;
    fn is_vectorized(self) -> bool
    where
        Self: Sized,
    {
        false
    }
}
