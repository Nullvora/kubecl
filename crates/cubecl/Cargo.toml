[package]
authors = ["nathanielsimard <nathaniel.simard.42@gmail.com>"]
categories = []
description = ""
edition.workspace = true
keywords = [
    "cuda",
    "wgpu",
    "gpgpu",
    "deep-learning",
    "machine-learning",
    "tensor",
]
license.workspace = true
name = "cubecl"
readme.workspace = true
repository = "https://github.com/tracel-ai/cubecl"
version.workspace = true
rust-version = "1.75"

[features]
default = ["cubecl-wgpu?/default", "cubecl-cuda?/default", "linalg"]
std = ["cubecl-wgpu?/std", "cubecl-cuda?/std"]
template = ["cubecl-core/template"]
linalg = ["dep:cubecl-linalg"]

# Runtimes
wgpu = ["cubecl-wgpu"]
cuda = ["cubecl-cuda"]

[dependencies]
cubecl-core = { path = "../cubecl-core", version = "0.1.0", default-features = false }
cubecl-wgpu = { path = "../cubecl-wgpu", version = "0.1.0", default-features = false, optional = true }
cubecl-cuda = { path = "../cubecl-cuda", version = "0.1.0", default-features = false, optional = true }
cubecl-linalg = { path = "../cubecl-linalg", version = "0.1.0", default-features = false, optional = true }

[[bench]]
name = "matmul"
harness = false
