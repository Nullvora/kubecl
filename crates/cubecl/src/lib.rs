pub use cubecl_core::*;

#[cfg(feature = "wgpu")]
pub use cubecl_wgpu as wgpu;

#[cfg(feature = "cuda")]
pub use cubecl_cuda as cuda;

#[cfg(feature = "lac")]
pub use cubecl_lac as lac;
