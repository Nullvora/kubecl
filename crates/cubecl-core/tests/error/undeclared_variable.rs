use cubecl_core as cubecl;
use cubecl::prelude::*;

#[cube]
fn kernel(x: UInt) {
    if x == y {
    }
}

fn main() {}
