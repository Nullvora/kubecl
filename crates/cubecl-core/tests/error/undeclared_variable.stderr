error: Variable not declared
 --> tests/error/undeclared_variable.rs:6:13
  |
6 |     if x == y {
  |             ^

error[E0425]: cannot find value `y` in this scope
 --> tests/error/undeclared_variable.rs:6:13
  |
6 |     if x == y {
  |             ^ help: a local variable with a similar name exists: `x`
